dynatrace/src/test/java/com/dynatrace/opentelemetry/metric/SerializerTest.java

```diff
@@ -32,31 +33,62 @@
       new Serializer(MetricBuilderFactory.builder().build());
 
   @Test
-  void fromLabels() {
-    Labels labels = Labels.of("label1", "value1", "label2", "value2");
+  void fromAttributes() {
+    Attributes attributes =
+        Attributes.builder().put("attr1", "value1").put("attr2", "value2").build();
+
     DimensionList expected =
         DimensionList.create(
-            Dimension.create("label1", "value1"), Dimension.create("label2", "value2"));
-    DimensionList actual = Serializer.fromLabels(labels);
+            Dimension.create("attr1", "value1"), Dimension.create("attr2", "value2"));
+
+    DimensionList actual = Serializer.fromAttributes(attributes);
 
     assertEquals(expected.getDimensions(), actual.getDimensions());
     assertNotSame(expected, actual);
   }
 
   @Test
-  void fromLabelsEmpty() {
-    Labels labels = Labels.empty();
+  void fromTypedAttributes() {
+    Attributes attributes =
+        Attributes.builder().put("l1", 1).put("l2", 1.5).put("l3", true).build();
+
     DimensionList expected = DimensionList.create();
-    DimensionList actual = Serializer.fromLabels(labels);
+
+    DimensionList actual = Serializer.fromAttributes(attributes);
 
     assertEquals(expected.getDimensions(), actual.getDimensions());
     assertNotSame(expected, actual);
   }
 
   @Test
-  void fromLabelsNormalize() {
-    Labels labels = Labels.of("~~!123", "test", "!!test2", "test2");
-    DimensionList actual = Serializer.fromLabels(labels);
+  void fromTypedAttributesArrayValues() {
+    Attributes attributes = Attributes.builder().put("l1", "l1v1", "l1v2").put("l2", 1, 2).build();
```

Comment:
Could also add one for booleans or doubles if you want, but i think clear from the context anyway.

Reply:
Done.